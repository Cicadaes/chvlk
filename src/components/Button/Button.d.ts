/// <reference types="react" />
import * as React from 'react';
export interface IButtonProps {
    onClick: (event: React.SyntheticEvent<HTMLButtonElement>) => void;
    children: JSX.Element | string;
}
export declare class Button extends React.Component<IButtonProps, undefined> {
    static displayName: string;
    render(): JSX.Element;
}
export default Button;
