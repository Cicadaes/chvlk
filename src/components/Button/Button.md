The one true button in UI Kit

```
<Button>Button</Button>
```