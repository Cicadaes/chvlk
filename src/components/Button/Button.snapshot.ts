import processSnapshots from '../../../utils/snapshots';
import Button, { IButtonProps } from './Button';

const snapshots = [{ props: { children: 'Button', onClick: jest.fn() } }];

processSnapshots<IButtonProps>(Button, snapshots);
