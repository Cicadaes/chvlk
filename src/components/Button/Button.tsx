import * as React from 'react';
import styled from '../../../utils/styled';
import { IStyledTheme } from '../../themes';

export interface IButtonProps {
  onClick: (event: React.SyntheticEvent<HTMLButtonElement>) => void;
  children: JSX.Element | string;
}

const ButtonElement = styled.button`
  font-size: 13px;
  line-height: 18px;
  color: ${({ theme }: { theme: IStyledTheme }): string => theme.colors.main};
  padding: 12px 16px;
  background: ${({ theme }: { theme: IStyledTheme }): string =>
    theme.colors.background};
  border: 1px solid ${({ theme }: { theme: IStyledTheme }): string =>
    theme.colors.main};
  cursor: pointer;

  :hover {
    background: ${({ theme }: { theme: IStyledTheme }): string =>
      theme.colors.main};
    color: ${({ theme }: { theme: IStyledTheme }): string =>
      theme.colors.background};
  }
`;

export class Button extends React.Component<IButtonProps, undefined> {
  public static displayName: string = 'Button';

  public render(): JSX.Element {
    const {} = this.props;
    return (
      <ButtonElement onClick={this.props.onClick}>
        {this.props.children}
      </ButtonElement>
    );
  }
}

export default Button;
