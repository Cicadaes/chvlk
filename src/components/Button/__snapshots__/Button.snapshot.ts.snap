// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshots 1`] = `
<styled.button
  onClick={[Function]}
>
  Button
</styled.button>
`;
